No member named 'to_time_t' in 'std::chrono::steady_clock'

Hydrogen version *: 2.0.0-pre-alpha-'168614c71'
Operation system + version: OpenBSD 7.8

---

Compilation fails with the following error on OpenBSD:
```
/tmp/hydrogen/src/core/Helpers/Time.cpp:40:19: error: no member named 'to_time_t' in 'std::chrono::steady_clock'
   40 |                 auto t = Clock::to_time_t( timePoint );
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -416,6 +416,21 @@ if(WIN32)
     set(CMAKE_COLOR_MAKEFILE OFF)
 endif()
 
+include(CheckCXXSourceCompiles)
+check_cxx_source_compiles([[
+  #include <chrono>
+  int main()
+  {
+    auto now = std::chrono::high_resolution_clock::now();
+    auto test = std::chrono::high_resolution_clock::to_time_t(now);
+    return 0;
+  }
+]] HAVE_TO_TIME_T)
+
+if(HAVE_TO_TIME_T)
+    add_compile_definitions(HAVE_TO_TIME_T=1)
+endif()
+
 #
 # CONFIG PROCESS SUMMARY
 #

--- a/src/core/Helpers/Time.cpp
+++ b/src/core/Helpers/Time.cpp
@@ -29,7 +29,7 @@
 
 namespace H2Core {
 	QString timePointToQString( const TimePoint& timePoint ) {
-#ifdef Q_OS_MACX
+#ifndef HAVE_TO_TIME_T
 		// We do not use `Clock` defined above since `high_resolution_clock` on
 		// macOS is an alias for `steady_clock`, which - in contrast to
 		// `system_clock` - does not have a `to_time_t` member.

